### PR TITLE
Append ExpressionAtlas gene link at the end of Json being returned to Reach component

### PR DESCRIPTION
**Pull request by haideriqbal**
Not much code edit required for this task just appended the Expression Atlas link at the end of json payload in JsonBioentityInformationController.java
Add new test in JsonBioentityInformationControllerWIT to test if payload contains ExpressionAtlasLink

Links to green builds:
http://bar:8085/browse/ATLAS-GXACIDEVTASKS7
http://bar:8085/browse/ATLAS-BASECIDEVTASKS7-1

### CODE REVIEW

File: sc/src/main/java/uk/ac/ebi/atlas/search/JsonBioentityInformationController.java [hunk 1]
```diff
@@ -55,6 +59,29 @@ public String getBioentityInformation(@PathVariable String geneId) {
                         geneName,
                         propertyValues);
 
-        return model.get("bioentityProperties").toString();
+
+        // Removed last bracket '[' from the return Json to append the AtlasUrl at the end of this Json
+        String bioentityProperties = model.get("bioentityProperties").toString();
```
Comment (alfonsomunozpomer) [inline sc/src/main/java/uk/ac/ebi/atlas/search/JsonBioentityInformationController.java]: Patching serialized representations of objects is dangerous and error prone. I strongly discourage you to do this. Start with `JsonArray jsonArray = GSON.fromJson(model.get("bioentityProperties").toString(), JsonArray.class);` and work with `jsonArray`.

Comment (haideriqbal) [reply sc/src/main/java/uk/ac/ebi/atlas/search/JsonBioentityInformationController.java]: Agreed. Will update the code according to suggestion given.

File: sc/src/main/java/uk/ac/ebi/atlas/search/JsonBioentityInformationController.java [hunk 2]
```diff
@@ -79,9 +76,8 @@ public String getBioentityInformation(@PathVariable String geneId) {
         o.addProperty("name", "ExpressionAtlas");
         o.add("values", values);
 
-        //Added a comma',' and end bracket '[' which was removed earlier to maintain valid Json format
-        String result = bioentityProperties + "," + GSON.toJson(o) + "]";
+        jsonArray.add(o);
 
-        return result;
+        return jsonArray.toString();
```
Comment (alfonsomunozpomer) [inline sc/src/main/java/uk/ac/ebi/atlas/search/JsonBioentityInformationController.java]: Use `GSON.toJson(jsonArray);`

Even if `toString` does what you’d expect it has the following downsides:
1. Maybe the contract is changed, `toString` could just well be the hash of the object.
2. Gson objects can be built with options to do lenient serialization, pretty printing (with indentations) and other options that aren’t possible with `toString`.

Remember: `Gson.fromJson` reads JSON, `Gson.toJson` writes JSON.

File: sc/src/test/java/uk/ac/ebi/atlas/search/JsonBioentityInformationControllerWIT.java [hunk 3]
```diff
@@ -12,9 +12,7 @@
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
 import org.springframework.web.context.WebApplicationContext;
 
-import static org.hamcrest.Matchers.greaterThanOrEqualTo;
-import static org.hamcrest.Matchers.hasSize;
-import static org.hamcrest.Matchers.isA;
+import static org.hamcrest.Matchers.*;
```
Comment (alfonsomunozpomer) [inline sc/src/test/java/uk/ac/ebi/atlas/search/JsonBioentityInformationControllerWIT.java]: It’s good practice to just bring the imports you need. Intent is clearer and it’s impossible you end up having conflicts.

**Pull request merged**